sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -748,6 +748,10 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
       values.map(value => s"$name = $value").mkString("(", " or ", ")")
     }
 
+    def convertNotInToAnd(name: String, values: Seq[String]): String = {
+      values.map(value => s"$name != $value").mkString("(", " and ", ")")
```

Comment:
More than 10,000 values will cause the Hive Metastore stack overflow.

Reply:
How about stoping push it If it's values size exceeds the threshold ? In this case it can not be covert like `>= and <=`.